fl4health/clients/moon_client.py

```diff
@@ -89,8 +89,6 @@ def get_contrastive_loss(
         return self.ce_criterion(logits, labels)
 
     def set_parameters(self, parameters: NDArrays, config: Config) -> None:
-        assert isinstance(self.model, MoonModel)
```

Comment:
This is unnecessary, unless I'm mistaken. Removing it also allows for FedPer models to be used with MOON Clients, which is nice...

Reply:
That's a fair question. The main reason I didn't do that was because FedPer model's exchange a partial subset of weights and don't, at least by default, admit projection modules for their features. They are very related. So it's possible that unifying them is a good idea. What do you think?

Reply:
projection_module is also optional in moon_base, so you can easily pass by None for it in inheritance, and everything should work well. I kinda prefer fedper_base to inherit from both moon and partial_layer_exchange_model so users can get the relationship between all of them and the added functionality of fed_per based on both.

Reply:
Yeah I think that makes sense. I'll do that right now.

Reply:
I added the assert back in now that FedPer inherits from MOON